src/D2L.CodeStyle.Analyzers/Immutability/ImmutabilityContext.Factory.cs

```diff
@@ -164,7 +164,11 @@ internal partial class ImmutabilityContext {
 				annotationsContext: annotationsContext,
 				extraImmutableTypes: extraImmutableTypesBuilder.ToImmutable(),
 				knownImmutableReturns: knownImmutableReturnsBuilder.ToImmutable(),
-				conditionalTypeParamemters: ImmutableHashSet<ITypeParameterSymbol>.Empty
+				conditionalTypeParamemters: ImmutableHashSet<ITypeParameterSymbol>.Empty,
+				regexInfo: (
+					GetTypeSymbol( compilationAssemblies, compilation, default, "System.Text.RegularExpressions.Regex" ),
+					GetTypeSymbol( compilationAssemblies, compilation, default, "System.CodeDom.Compiler.GeneratedCodeAttribute" )
+				)

```

Comment:
We [consider `System.Text.RegularExpressions.Regex` immutable](https://github.com/Brightspace/D2L.CodeStyle/blob/32cd025d84490929ffb16584aa125406e3b77c6b/src/D2L.CodeStyle.Analyzers/Immutability/ImmutabilityContext.Factory.cs#L54) already